felix/docker-image/Dockerfile

```diff
@@ -67,7 +67,7 @@ RUN update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
 
 # Felix is built with RHEL/UBI and links against libpcap.so.1. We need this symbolic link
 # until Debian changes the soname from .0.8 to .1.
-RUN ln -s $(readlink /usr/lib/*-linux-gnu/libpcap.so.0.8) /usr/lib/x86_64-linux-gnu/libpcap.so.1
+RUN ln -s $(readlink /usr/lib/*-linux-gnu/libpcap.so.0.8) $(ldconfig -v 2>/dev/null | grep '^/lib/' | cut -f1 -d:)/libpcap.so.1
```

Comment:
Thanks for the explanations, which I've reviewed.  I'm afraid I'm still unsure on a couple of points.

1. As it stands, `$(ldconfig ...)` will generate multiple lines (or words) of output, whereas I think we only want 1.  In the original incantation there was a `head -n 1` to trim down to just one output line.
2. I'm unclear which library directory we're trying to extract from the ldconfig output, and hence if the `^` is really wanted.  If the desired answer on some platforms is `/usr/lib/x86_64-linux-gnu`, then I was wrong to recommend the `^`.  Running on my laptop, the results without the `^` are:
```
/usr/lib/x86_64-linux-gnu/libfakeroot
/lib/i386-linux-gnu
/usr/local/lib
/lib/x86_64-linux-gnu
/lib
/lib/i386-linux-gnu/sse2
```
and with the `^`:
```
/lib/i386-linux-gnu
/lib/x86_64-linux-gnu
/lib
/lib/i386-linux-gnu/sse2
```

Reply:
@nelljerram My understanding is a bit limited, but I don't think it matters much whether it is in /usr/lib or /lib for our purposes. Won't both locations be searched when linking?

For the amd64 debian:12-slim image at least, there is no difference whether or not we restrict to `^/lib` or `/lib` - they both return a single element.

Agree `head -n 1` should be added back. I think in my initial testing of the new command I was only getting a single result anyway, so forgot to include it.

Reply:
OK, so your thinking is that it doesn't matter where we create the link, so long as it's one of the places that ld.so is going to look in.  That makes sense.  Will approve now and leave any final tweaks up to you!

Reply:
Thanks for the quick and thorough review, @nelljerram!